File: app/src/main/java/com/fastaccess/data/dao/AuthModel.java
package com.fastaccess.data.dao;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

import java.util.List;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * Created by Kosh on 12 Mar 2017, 3:16 AM
 */

@Getter @Setter @NoArgsConstructor
public class AuthModel implements Parcelable {

    private String clientId;
    private String clientSecret;
    private String redirectUri;
    private List<String> scopes;
    private String state;
    private String note;
    private String noteUrl;
    @SerializedName("X-GitHub-OTP") private String otpCode;

    @Override public int describeContents() { return 0; }

    @Override public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.clientId);
        dest.writeString(this.clientSecret);
        dest.writeString(this.redirectUri);
        dest.writeStringList(this.scopes);
        dest.writeString(this.state);
        dest.writeString(this.note);
        dest.writeString(this.noteUrl);
        dest.writeString(this.otpCode);
    }

    private AuthModel(Parcel in) {
        this.clientId = in.readString();
        this.clientSecret = in.readString();
        this.redirectUri = in.readString();
        this.scopes = in.createStringArrayList();
        this.state = in.readString();
        this.note = in.readString();
        this.noteUrl = in.readString();
        this.otpCode = in.readString();
    }

    public static final Creator<AuthModel> CREATOR = new Creator<AuthModel>() {
        @Override public AuthModel createFromParcel(Parcel source) {return new AuthModel(source);}

        @Override public AuthModel[] newArray(int size) {return new AuthModel[size];}
    };
}

File: network/src/main/java/com/fastaccess/domain/response/AuthModel.kt
package com.fastaccess.domain.response

import com.google.gson.annotations.SerializedName

/**
 * Created by Kosh on 11.05.18.
 */
data class AccessTokenResponse(var id: Long? = null,
                               var token: String? = null,
                               var hashedToken: String? = null,
                               var accessToken: String? = null,
                               var tokenType: String? = null)

data class AuthBodyModel(var clientId: String? = null,
                         var clientSecret: String? = null,
                         var redirectUri: String? = null,
                         var scopes: List<String>? = null,
                         var state: String? = null,
                         var note: String? = null,
                         var noteUrl: String? = null,
                         @SerializedName("X-GitHub-OTP") var otpCode: String? = null)